sentry-spring/src/main/java/io/sentry/spring/reactive/SentryReactiveWebFilter.java

```diff
@@ -0,0 +1,52 @@
+package io.sentry.spring.reactive;
+
+import com.jakewharton.nopen.annotation.Open;
+import io.sentry.Breadcrumb;
+import io.sentry.IHub;
+import io.sentry.SentryOptions;
+import io.sentry.util.Objects;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.core.Ordered;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.web.server.ServerWebExchange;
+import org.springframework.web.server.WebFilter;
+import org.springframework.web.server.WebFilterChain;
+import reactor.core.publisher.Mono;
+
+/**
+ * Associates a cloned {@link io.sentry.IHub} to each incoming reactive HTTP request
+ * (ServerWebExchange).
+ */
+@Open
+public class SentryReactiveWebFilter implements WebFilter, Ordered {
+
+  private final @NotNull IHub baseHub;
+  private final @NotNull SentryOptions options;
+
+  public SentryReactiveWebFilter(final @NotNull IHub hub, final @NotNull SentryOptions options) {
+    this.baseHub = Objects.requireNonNull(hub, "hub is required");
+    this.options = Objects.requireNonNull(options, "options are required");
+  }
+
+  @Override
+  public Mono<Void> filter(ServerWebExchange exchange, WebFilterChain chain) {
+    final IHub hub = baseHub.clone();
+    hub.pushScope();
```

Comment:
https://github.com/getsentry/sentry-java/blob/main/sentry-spring/src/main/java/io/sentry/spring/SentrySpringRequestListener.java

do a pop scope once its over, I believe we need something similar here

Reply:
If the scope is within this method, then a `try/finally` would be ideal. If we don't pop, things will get messy

Reply:
Thanks for your suggestions, I'll put it within a try/finally, but its worth to note that the Hub is cloned in every single request since it can not be associated to a Thread Local variable because of the Thread model of Webflux

Reply:
That raises the question if we need to `push` at all then. But I'd suggest we continue with the `push` and `pop` per request. Thanks for adding it. For example, we'll add performance monitoring support soon and one of the ways we could (not sure how it'll be done yet) flush the transaction, is when a scope if popped.